poppy/utils.py

```diff
@@ -1601,32 +1604,32 @@ def sort_sptype(typestr):
 
     if "Flat" in sptype:
         if sptype == "Flat spectrum in F_nu":
-            spec = pysynphot.FlatSpectrum(1, fluxunits='fnu')
+            spec = SourceSpectrum(ConstFlux1D, amplitude=1 * syn_u.FNU)
         elif sptype == "Flat spectrum in F_lambda":
-            spec = pysynphot.FlatSpectrum(1, fluxunits='flam')
-        spec.convert('flam')
-        return spec * (1. / spec.flux.mean())
+            spec = SourceSpectrum(ConstFlux1D, amplitude=1 * syn_u.FLAM)
+
+        return spec
     if 'Power law' in sptype:
         import re
-        ans = re.search('\((.*)\)', sptype)
+        ans = re.search(r'\((.*)\)', sptype)
         if ans is None:
             raise ValueError("Invalid power law specification cannot be parsed to get exponent")
         exponent = float(ans.groups(0)[0])
-        # note that Pysynphot's PowerLaw class implements a power law in terms of lambda, not nu.
+        # note that synphot's PowerLaw class implements a power law in terms of lambda, not nu.
```

Comment:
It is now possible to define powerlaw in frequency space but that functionality isn't widely tested.